app/DynamicsAdapter/Fams3Adapter.Dynamics/SearchRequest/SearchRequestService.cs

```diff
@@ -659,5 +663,30 @@ private async Task<EmploymentEntity> LinkEmploymentRef(EmploymentEntity employme
             employment.CountrySubdivision = await GetEmploymentSubdivision(employment.CountrySubdivisionText, cancellationToken);
             return employment;
         }
-    }
+
+        private async Task<SSG_Person> FindDuplicatedPerson(PersonEntity newPerson)
+        {
+            newPerson.DuplicateDetectHash = await _duplicateDetectService.GetDuplicateDetectHashData(newPerson);
+            string hashData = newPerson.DuplicateDetectHash;
+            try
+            {
+                var existedPerson = await this._oDataClient.For<SSG_Person>()
+                    .Filter(x => x.DuplicateDetectHash == hashData)
+                    .FindEntryAsync(CancellationToken.None);
+                    
+                return existedPerson;
+            }
+            catch (WebRequestException e)
```

Comment:
Can we use exception filter here  `catch (WebRequestException e) when (e.Code == HttpStatusCode.NotFound)`

Reply:
agree. code changed.